### ISSUE TITLE
Mirrored Posts on Twitter link back to Friendica, but when following the link Twitter users end up with a 404 error

### ISSUE THREAD

**Issue by foss-** [friendica/friendica#6853]
Long posts which are mirrored to Twitter end up in a shorter version followed by a link back to the Friendica post. This bug is known and has been happening for a while. Maybe not top priority, but should be fairly high. Getting Twitter users to follow posts back to friendica is a great way to raise awareness for that alternative social platform and get new users interested in it.

I think it's really important the first user experience potential twitter switchers have is not a 404 error.

### Expected behavior

Working link back to Friendica.

### Actual behavior

404 error deadlock.

### Steps to reproduce the problem

Post a long (longer than the chars allowed for a single Tweet) post on friendica with Twitter mirror enabled. Post will show on Twitter with link back to friendica. Log-out of friendica and visit link from Twitter. -> 404

**Comment by MrPetovan** [comment]
Can you please post a link to both the Twitter post and the expected Friendica post link when you're logged in? Could be a permission issue where the connector wrongly mirrored a restricted post, which would explain the 404 error.

**Comment by annando** [comment]
The problem is based upon how we treat imported Twitter posts. Although they are (mostly) public we disallow access from someone else. Comments inherit the permissions from the starting post. So your comment isn't available to the public as well.

Question is how to treat it.

**Comment by MrPetovan** [comment]
No, it's the other way around, Friendica -> Twitter. Since it's a post longer than 280 chars, it is truncated and a link to the original Friendica post is added. Apparently the link to the post shows a 404, which might indicate that the post is restricted.

**Comment by annando** [comment]
The case is described here: https://forum.friendi.ca/display/39bbe52a-185c-864c-33e9-3ae852365887

it is an answer to a tweet.

**Comment by MrPetovan** [comment]
Ah, sorry, I see that now. The best solution would be a public post that can only be interacted with by the Twitter connector importer. Can we do that?

**Comment by annando** [comment]
What do you mean with "can only be interacted with by the Twitter connector importer"?

We possibly should simply make public Twitter posts public.

**Comment by MrPetovan** [comment]
But if they are public, wouldn't your Friendica contacts be able to interact with it? Would it make sense? I seem to remember you told me this was the reason why connector posts were private, because they only make sense to the importer.

**Comment by annando** [comment]
Ah, now I understand. We had to separate between public readable and interactivable. The last one can't be done - of course.

**Comment by MrPetovan** [comment]
Just "interactive". The word existed.

**Comment by annando** [comment]
Pöh! 👅

**Comment by Extarys** [comment]
@foss- Is this the error I've been experiencing?
![image](https://user-images.githubusercontent.com/11581433/106526785-2f933700-64b4-11eb-8c63-28f5dfc47b3f.png)

**Comment by MrPetovan** [comment]
Possibly, it depends on where you got the page URL.

**Comment by Extarys** [comment]
That link: https://pirati.ca/display/ec054ce7-5460-16de-dc7b-7c4231346227

Foss- sent it to me so I could read the post and I got that error so we were looking through GH issues. No idea how I could log/debug this though.

**Comment by MrPetovan** [comment]
And this was the URL of a Friendica reply on a public Twitter post?

**Comment by foss-** [comment]
Good point, it was a friendica post going to https://venera.social/display/85a863ed-7860-1485-a394-05a916453658 found on piratica as per https://pirati.ca/display/85a863ed-7860-1485-a394-05a916453658

So this case is not identical but similar and might have the same cause.
